deploy/deploy_mxbluesky.py

```diff
@@ -0,0 +1,189 @@
+"""
+Deploy latest release of mx_bluesky either on a beamline or in dev mode.
+"""
+
+import argparse
+import os
+from subprocess import PIPE, CalledProcessError, Popen
+from typing import List, Tuple
+
+from git import Repo
+from packaging.version import Version
+
+recognised_beamlines = ["i24"]
+
+help_message = """
+To deploy mx_bluesky on a specific beamline, pass only the --beamline argument.
+This will put the latest release in /dls_sw/ixx/software/bluesky/mx_bluesky_v#.#.# and \
+set the permissions accordingly. \n
+To run in dev mode instead, only the --dev-path should be passed, a test release will \
+be placed in {dev_path}/mxbluesky_release_test/bluesky. \n
+Finally, if both a --beamline and a --dev-path are specified, a beamline-specific test \
+deployment will be put in the test directory.
+"""
+
+
+class repo:
+    # Set name, setup remote origin, get the latest version"""
+    def __init__(self, name: str, repo_args):
+        self.name = name
+        self.repo = Repo(repo_args)
+
+        self.origin = self.repo.remotes.origin
+        self.origin.fetch()
+
+        self.versions = [t.name for t in self.repo.tags]
+        self.versions.sort(key=Version, reverse=True)
+        print(f"Found {self.name}_versions:\n{os.linesep.join(self.versions)}")
+        self.latest_version_str = self.versions[0]
+
+    def deploy(self, url, beamline: str = None):
+        print(f"Cloning latest version {self.name} into {self.deploy_location}")
+
+        deploy_repo = Repo.init(self.deploy_location)
+        deploy_origin = deploy_repo.create_remote("origin", self.origin.url)
+
+        deploy_origin.fetch()
+        deploy_repo.git.checkout(self.latest_version_str)
+
+        print("Setting permissions")
+        groups_to_give_permission = get_permission_groups(beamline)
+        setfacl_params = ",".join(
+            [f"g:{group}:rwx" for group in groups_to_give_permission]
+        )
+
+        # Set permissions and defaults
+        os.system(f"setfacl -R -m {setfacl_params} {self.deploy_location}")
+        os.system(f"setfacl -dR -m {setfacl_params} {self.deploy_location}")
+
+    # Deploy location depends on the latest mx_bluesky version (...software/bluesky/mx_bluesky_V...)
+    def set_deploy_location(self, release_area):
+        self.deploy_location = os.path.join(release_area, self.name)
+        if os.path.isdir(self.deploy_location):
+            raise Exception(
+                f"{self.deploy_location} already exists, stopping deployment for {self.name}"
+            )
+
+
+# Get permission groups depending on beamline/dev install
+def get_permission_groups(beamline: str = None) -> List:
+    if not beamline:
+        return ["gda2", "dls_dasc"]
+    return [f"{beamline}_staff", "gda2", "dls_dasc"]
```

Comment:
Nit: Avoids a bit of repetition
```suggestion
    beamline_groups = ["gda2", "dls_dasc"]
    if beamline:
        beamline_groups.append(f"{beamline}_staff")
    return beamline_groups
```